### PR TITLE
Only markdown #1660

### PR DESCRIPTION
**Pull request by cbeauchesne**
Sorry for the mess...

### CODE REVIEW

**Comment by cbeauchesne** [comment]
Ok, as I was saying, before putitng a damn mess everywhere : 

Ok, I'm ready for merging.

But before getting in prod, there is two mandatory things : 

* a demo server with an API with pretty recent DB (actual API data are far too old) for testing.
* and a DB dump, because I need to prepare the bot migration scripts.

I can merge now, but if we can't have this two things on time, it's better to wait IMO.

**Comment by cbeauchesne** [comment]
For the merge, can I use Squash and merge, or is there a better option?

File: Dockerfile_dev_environment [hunk 1]
```diff
@@ -26,13 +26,14 @@ RUN set -x \
...
-    nodejs-legacy \
-    npm \
     openjdk-7-jdk \
     gettext \
     gcc \
     git \
     virtualenv \
+    curl \
+ && curl -sL https://deb.nodesource.com/setup_8.x | bash - \
+ && apt-get install -y nodejs \
```
Comment (asaunier) [inline Dockerfile_dev_environment]: Are those changes really related to the parser changes? Could you explain what they are used for?

Comment (cbeauchesne) [reply Dockerfile_dev_environment]: Not mine, it's from bruno : https://github.com/c2corg/v6_ui/pull/1844/commits/e770aba3fc62c2f520ca7941bb8d74ab7c318db8

I can't say if it's normal to have it in this PR or not.

I'll take the time to learn git, and have organized commits in the future, but I appreciate the exception for this one, i spent more time trying to organize them, than coding. Quite a pain...

Comment (brunobesson) [reply Dockerfile_dev_environment]: I think this is a problem with rebase g, because these modifications have been made by myself to correct the Docker build which requires a more recent version of node

Comment (asaunier) [reply Dockerfile_dev_environment]: Indeed, this change is already available in the master branch: https://github.com/c2corg/v6_ui/commit/7c3c0e7e93593becd8fe42676df65dc1d82520a1
It's a bit strange git does not report a conflict?!
I will try to propose a cleaned (separated) branch. I let you know.

Comment (cbeauchesne) [reply Dockerfile_dev_environment]: Thank you :)

File: c2corg_ui/tests/format/test_format.py [hunk 2]
```diff
@@ -9,6 +9,9 @@
 
 
 class TestFormat(unittest.TestCase):
+
+    maxDiff = None
```
Comment (asaunier) [inline c2corg_ui/tests/format/test_format.py]: Is this change really needed? This property seems to be use nowhere else...

File: c2corg_ui/format/img.py [hunk 3]
```diff
@@ -15,28 +15,28 @@
 
 
 class C2CImageExtension(Extension):
-
```
Comment (asaunier) [inline c2corg_ui/format/img.py]: Why removing this blank line? I think in all class definitions we have 1 blank line before every method. I am a bit surprised the flake8 makefile target does not raise an error for this.

File: c2corg_ui/format/video.py [hunk 4]
```diff
@@ -14,34 +14,48 @@
 
 
 class C2CVideoExtension(Extension):
+    def __init__(self, *args, **kwargs):
```
Comment (asaunier) [inline c2corg_ui/format/video.py]: Missing blank line

File: c2corg_ui/format/img.py [hunk 5]
```diff
@@ -15,28 +15,28 @@
...
+        pattern = C2CImage(IMG_RE, self.getConfigs(), self._ngclick_secret_tag)
         pattern.md = md
         # append to end of inline patterns
         md.inlinePatterns.add('c2cimg', pattern, "<not_strong")
 
 
 class C2CImage(Pattern):
-
-    def __init__(self, pattern, config):
+    def __init__(self, pattern, config, ngclick_secret_tag):
```
Comment (asaunier) [inline c2corg_ui/format/img.py]: Please keep the blank line

File: c2corg_ui/format/video.py [hunk 6]
```diff
@@ -14,34 +14,48 @@
...
-        pattern = C2CVideo(VIDEO_RE, markdown_instance=md)
+        pattern = C2CVideo(VIDEO_RE, markdown_instance=md,
+                           iframe_secret_tag=self._iframe_secret_tag)
         pattern.md = md
         # append to end of inline patterns
         md.inlinePatterns.add('c2cvideo', pattern, "<extra_autolink")
 
 
 class C2CVideo(Pattern):
+    def __init__(self, *args, **kwargs):
```
Comment (asaunier) [inline c2corg_ui/format/video.py]: Missing blank line

File: c2corg_ui/format/alerts.py [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+    def extendMarkdown(self, md, md_globals):  # noqa
+        # md.registerExtension(self)
+
+        md.parser.blockprocessors.add('c2calert',
+                                      AlertProcessor(md.parser),
+                                      "<paragraph")
+
+
+def makeExtension(*args, **kwargs):  # noqa
+    return AlertExtension(*args, **kwargs)
```
Comment (asaunier) [inline c2corg_ui/format/alerts.py]: The ``AlertExtension`` is defined but not used (not included in https://github.com/c2corg/v6_ui/blob/OnlyMarkdown/c2corg_ui/format/__init__.py#L107-L119). What's the point?

Are there tests provided for this extension?

I recommend to remove this file from this PR and to add it in a dedicated PR.

**Comment by asaunier** [comment]
> The easiest solution is see is to start from a clean and up-to-date copy of the master branch and apply the changes by hand by copying-pasting the changed files. I will try to do it.

Actually I won't because doing so would made me the author of the changes which would not be fair. @cbeauchesne could you do it?

What you have to do:
* save a copy of the v6_ui directory with the changes of the current branch
* go back to branch master and make sure it is up-to-date:
```
git checkout master
git pull
```
* create a new branch: ``git checkout -b OnlyMarkdown2``
* copy the changed files to the new branch without the Docker files changes (already available in master) and the ``AlertExtension`` file since it is not used yet (a dedicated PR would make sense).
* make a single commit with a short but explicit title
* push the branch and make a new PR.

The other option is to squash and merge the current PR but I don't really know what will happen (I guess it will create a single commit with the Docker files change and the AlertExtension as visible in the diff) but I am really not sure (there are plenty of commits in the middle of the suggested commits - have a look at the output of "git log").

Thanks!

**Comment by asaunier** [comment]
@cbeauchesne Please close this PR since it has been superseeded by #1932 
Don't forget to delete the branch as well (click the "delete branch" button that should appear after closing).
